deepchem/molnet/load_function/material_datasets/load_bandgap.py

```diff
@@ -0,0 +1,201 @@
+from deepchem.trans import Transformer
+from deepchem.splits.splitters import Splitter
+from deepchem.molnet.defaults import get_defaults
+
+from typing import List, Tuple, Dict, Optional, Union, Any, Type
+
+logger = logging.getLogger(__name__)
+
+# TODO: Change URLs
+DEFAULT_DIR = deepchem.utils.get_data_dir()
+BANDGAP_URL = 'http://deepchem.io.s3-website-us-west-1.amazonaws.com/datasets/expt_gap.tar.gz'
+
+# dict of accepted featurizers for this dataset
+# modify the returned dicts for your dataset
+DEFAULT_FEATURIZERS = get_defaults("feat")
+
+# Names of supported featurizers
+featurizers = [
+    'ElementPropertyFingerprint',
+]
+DEFAULT_FEATURIZERS = {k: DEFAULT_FEATURIZERS[k] for k in featurizers}
+
+# dict of accepted transformers
+DEFAULT_TRANSFORMERS = get_defaults("trans")
+
+# dict of accepted splitters
+DEFAULT_SPLITTERS = get_defaults("splits")
+
+# names of supported splitters
+splitters = ['RandomSplitter']
+DEFAULT_SPLITTERS = {k: DEFAULT_SPLITTERS[k] for k in splitters}
+
+
+def load_bandgap(
+    featurizer=DEFAULT_FEATURIZERS['ElementPropertyFingerprint'],
+    transformers: List = [DEFAULT_TRANSFORMERS['NormalizationTransformer']],
+    splitter=DEFAULT_SPLITTERS['RandomSplitter'],
+    reload: bool = True,
+    data_dir: Optional[str] = None,
+    save_dir: Optional[str] = None,
+    featurizer_kwargs: Dict[str, Any] = {'data_source': 'matminer'},
+    splitter_kwargs: Dict[str, Any] = {
+        'frac_train': 0.8,
+        'frac_valid': 0.1,
+        'frac_test': 0.1
+    },
+    transformer_kwargs: Dict[str, Dict[str, Any]] = {
+        'NormalizationTransformer': {
+            'transform_X': True
+        }
+    },
+    **kwargs) -> Tuple[List, Tuple, List]:
+  """Load band gap dataset.
+
+  Contains 4604 experimentally measured band gaps for inorganic
+  crystal structure compositions. In benchmark studies, random forest 
+  models achieved a mean average error of 0.45 eV during five-fold 
+  nested cross validation on this dataset. 
+
+  For more details on the dataset see [1]_. For more details
+  on previous benchmarks for this dataset, see [2]_.
+  
+  Parameters
+  ----------
+  featurizer : MaterialCompositionFeaturizer 
+    (default ElementPropertyFingerprint)
+    A featurizer that inherits from deepchem.feat.Featurizer.
+  transformers : List[Transformer]
+    A transformer that inherits from deepchem.trans.Transformer.
+  splitter : Splitter (default RandomSplitter)
+    A splitter that inherits from deepchem.splits.splitters.Splitter.
+  reload : bool (default True)
+    Try to reload dataset from disk if already downloaded. Save to disk
+    after featurizing.
+  data_dir : str, optional
+    Path to datasets.
+  save_dir : str, optional
+    Path to featurized datasets.
+  featurizer_kwargs : Optional[Dict[str, Any]]
+    Specify parameters to featurizer, e.g. {"size": 1024}
+  splitter_kwargs : Optional[Dict[str, Any]]
+    Specify parameters to splitter, e.g. {"seed": 42}
```

Comment:
In this case, you removed Optional from the type annotations but you still list it in the docstrings.